### PR TITLE
Add Iris middleware

### PR DESCRIPTION
**Pull request by kataras**
All tests are passed.

### CODE REVIEW

**Comment by kataras** [comment]
Hello @iamemilio, do you have any news regrading this one?

File: .travis.yml [hunk 1]
```diff
@@ -227,7 +227,10 @@ matrix:
     env:
       - DIRS=v3/integrations/nrgraphqlgo,v3/integrations/nrgraphqlgo/example
       - EXTRATESTING="go get -u github.com/graphql-go/graphql@master"
-
+  - go: "1.19"
+    env:
+      - DIRS=v3/integrations/nriris
+      - EXTRATESTING="go get -u github.com/kataras/iris/v12@master"
```
Comment (iamemilio) [inline .travis.yml:233]:
```suggestion
```
we don't use this, and are going to remove it soon

File: v3/integrations/nriris/nriris_test.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+	router.Use(Middleware(app.Application))
+	router.Get("/anon", func(ctx iris.Context) {
+		ctx.WriteString("anonymous function handler")
+	})
+
+	if err := router.Build(); err != nil {
+		t.Fatal(err)
+	}
+
+	txnName := "GET " + pkg + ".TestAnonymousHandler.func1"
```
Comment (iamemilio) [inline v3/integrations/nriris/nriris_test.go:95]:
```suggestion
	txnName := "GET " + pkg + ".TestAnonymousHandler"
```

File: v3/integrations/nriris/go.mod [hunk 3]
```diff
@@ -0,0 +1,61 @@
+module github.com/newrelic/go-agent/v3/integrations/nriris
+
+go 1.19
+
+require (
+	github.com/kataras/iris/v12 v12.2.0-beta6.0.20221105124545-5b80c297dccd
+	github.com/newrelic/go-agent/v3 v3.20.0
```
Comment (iamemilio) [inline v3/integrations/nriris/go.mod:7]:
```suggestion
	github.com/newrelic/go-agent/v3 v3.20.2
```

**Comment by iamemilio** [comment]
Sorry for the nits, I wasn't able to make the changes directly to this PR.  Overall, this looks good. The one thing I noticed was that requests that 404 did not show up in the monitoring when running the example code. I don't have a good understanding of the iris framework, and am not sure if that is by design. I figured you might want to know about that. If you think that is ok, then once these nits are resolved, I'm ready to move forward on this.

**Comment by kataras** [comment]
Hello @iamemilio, thansk a lot for the review! It's ok because the monitor middleware was registered on existing routes, if we want to catch 404s too we can change it to `UseRouter` instead of `Use`. I have commited your changes as you proposed.

**Comment by iamemilio** [comment]
Apologies, but we deleted the `travis.yml` and it created a merge conflict. Once that is resolved, this is approved to merge.

**Comment by iamemilio** [comment]
@kataras one more patch :)

**Pull request closed without merging**